src/System.Collections.Immutable/src/System.Collections.Immutable.csproj

```diff
@@ -43,7 +43,7 @@
     <Compile Include="System\Collections\Generic\ISortKeyCollection.cs" />
     <Compile Include="System\Collections\Immutable\AllocFreeConcurrentStack.cs" />
     <Compile Include="System\Collections\Immutable\DictionaryEnumerator.cs" />
-    <Compile Include="System\Collections\Immutable\DisposableEnumeratorAdapter`2.cs" />
+    <Compile Include="System\Collections\Immutable\DisposableEnumeratorAdapter.Generic.cs" />
```

Comment:
I don't have a strong opinion here, but I'm wondering if it'd be better to just remove the tick (e.g. DisposableEnumeratorAdapter2.cs), or replace it with an underscore (e.g. DisposableEnumeratorAdapter_2.cs).  Replacing the ```'2``` with ```.Generic``` means there's some information lost when reading the file name (you don't know the arity of the type in question), though that may not be a big deal.  I'm fine with whatever you and @AArnott decide here.